keras_core/export/export_lib.py

```diff
@@ -100,13 +107,25 @@ def non_trainable_variables(self):
 
     def track(self, resource):
         """Track the variables (and other assets) of a layer or model."""
-        if not isinstance(resource, tf.__internal__.tracking.Trackable):
+        if backend.backend() == "tensorflow" and not isinstance(
+            resource, tf.__internal__.tracking.Trackable
+        ):
             raise ValueError(
                 "Invalid resource type. Expected an instance of a "
                 "TensorFlow `Trackable` (such as a Keras `Layer` or `Model`). "
                 f"Received instead an object of type '{type(resource)}'. "
                 f"Object received: {resource}"
             )
+        if backend.backend() == "jax" and not isinstance(
+            resource, backend.jax.layer.JaxLayer
+        ):
+            raise ValueError(
+                "Invalid resource type. Expected an instance of a "
+                "JAX-based Keras `Layer` or `Model` "
+                "(backend.jax.layer.JaxLayer). "
```

Comment:
This is private, let's not mention it in the error message

Reply:
Removed, thanks.